app/controller_services/shopping_list_items_controller/create_service.rb

```diff
@@ -74,7 +74,7 @@ def all_matching_list_items
     end
 
     def lists_to_be_changed
-      list_ids = if params[:unit_weight] && params[:unit_weight] != aggregate_list_item&.unit_weight
+      list_ids = if all_matching_list_items.count > 0 && params[:unit_weight] && params[:unit_weight] != aggregate_list_item&.unit_weight
```

Comment:
I tried a couple approaches to making this line shorter but one caused a syntax error and the other caused Rubocop to do something really weird with the spacing, so I am reluctantly leaving this line very long.